### DISCUSSION TITLE
Client-side actions in form function are called after the form finishes submitting?

### DISCUSSION THREAD

**Discussion by vedantroy** [vercel/next.js#50551] [Help]
### Summary

I notice that when doing a client-side action inside of a form action, the client side action will only trigger *after* the server side action has completed. This is even if the client side action comes before the server side action in the code.

This is unexpected to me. Has anyone else experienced this?

### Additional information

```js
    <form action={(formData) => {
      if (inputText === "") return
      // For some reason this only clears the input after submission
      setInputText("")
      // server action
      return submitMessage(formData)
    }} className="relative flex-grow">
      <TextFormControls game={game} chat={chat} inputText={inputText} setInputText={setInputText}/>
    </form>
```

**Comment by icyJoseph** [comment]
I guess you should use `onSubmit` there, rather than action. The action prop is used to create a "bind" between the client and server, so that when the form submits, the server side action is called (by flying a request from the client).

If you want to use interactivity, use `onSubmit`.

Do remember that this in alpha stage, so it might as well change upside down tomorrow, or next week, until marked as beta or stable.
